core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1755,6 +1762,40 @@ class ReplicaManager(val config: KafkaConfig,
     partitionsToMakeFollower
   }
 
+  private def updateTopicIdForFollowers(controllerId: Int,
+                                        controllerEpoch: Int,
+                                        partitionStates: Set[Partition],
+                                        correlationId: Int,
+                                        topicIds: String => Option[Uuid]): Unit = {
+    val traceLoggingEnabled = stateChangeLogger.isTraceEnabled
+
+    try {
+      if (isShuttingDown.get()) {
+        if (traceLoggingEnabled) {
+          partitionStates.foreach { partition =>
+            stateChangeLogger.trace(s"Skipped the update topic ID step of the become-follower state " +
+              s"change with correlation id $correlationId from controller $controllerId epoch $controllerEpoch for " +
+              s"partition ${partition.topicPartition} since it is shutting down")
+          }
+        }
+      } else {
+        val partitionsToUpdateFollowerWithLeader = partitionStates.map { partition =>
+          val leaderNode = partition.leaderReplicaIdOpt.flatMap(leaderId => metadataCache.
+            getAliveBrokerNode(leaderId, config.interBrokerListenerName)).getOrElse(Node.noNode())
+          val leaderId = leaderNode.id
```

Comment:
So, you take the leader id from the partition, then you lookup the node from the metadata cache, finally you get the leader id from the node. I guess you could use the leader id directly. :)

Initially, I thought that you would put the `metadataCache.hasAliveBroker(partitionState.leader)` check here instead of doing it at L1407. It does not matter much but doing it here would be a bit more aligned with the logic in the `makeFollowers` method.

Reply:
Ah I see. I thought I copied the check from makeFollowers, but I guess I didn't. And I thought I removed the extra work to get the ID, but looks like there was still an unnecessary step. 😅